.github/workflows/scorecards.yml

```diff
@@ -0,0 +1,60 @@
+name: Scorecards supply-chain security
+on:
+  # Only the default branch is supported.
+  branch_protection_rule:
+  schedule:
+    - cron: '26 0 * * 0'
+  push:
+    branches: [ "main" ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecards analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Used to receive a badge.
+      id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846 # tag=v3.0.0
```

Comment:
is there a reason why we're pinning all these actions to specific commits and not tags?

Reply:
It's more secure to pin digests

Reply:
yes, the academic answer is pin to digest because tags can change, but commits also disappear when you rewrite git history

Reply:
Probably less likely the git history gets rewritten than a tag gets changed. Since we're already using the tag elsewhere I think we could use it here as well though I don't really mind either way.

Reply:
I think because this is a security related action, it makes sense to use the digest. If the git history is changed maliciously that means we will get a failure running the job that we can investigate. If a tag is changed maliciously then we would not see any failure.